deepchem/models/torch_models/infograph.py

```diff
@@ -101,81 +291,138 @@ class InfoGraphModel(ModularTorchModel):
     >>> ds = NumpyDataset(X, y, w)
     >>> num_feat = max([ds.X[i].num_node_features for i in range(len(ds))])
     >>> edge_dim = max([ds.X[i].num_edge_features for i in range(len(ds))])
-    >>> model = InfoGraphModel(num_feat, edge_dim, 15, use_unsup_loss=True, separate_encoder=True)
+    >>> model = InfoGraphStarModel(num_feat, edge_dim, 15, training_mode='semisupervised')
     >>> loss = model.fit(ds, nb_epoch=1)
     """
 
     def __init__(self,
                  num_features,
                  edge_features,
-                 dim,
-                 use_unsup_loss=False,
-                 separate_encoder=False,
+                 embedding_dim,
+                 training_mode='supervised',
                  measure='JSD',
                  average_loss=True,
+                 num_gc_layers=5,
                  **kwargs):
-        self.embedding_dim = dim
+
         self.edge_features = edge_features
-        self.separate_encoder = separate_encoder
         self.local = True
+        self.prior = False
+        self.gamma = .1
         self.num_features = num_features
-        self.use_unsup_loss = use_unsup_loss
         self.measure = measure
         self.average_loss = average_loss
+        self.training_mode = training_mode
+        if training_mode == 'supervised':
+            self.embedding_dim = embedding_dim
+            self.use_unsup_loss = False
```

Comment:
We should remove `self.use_unsup_loss`. It's no longer an argument and is a duplicate of information in `training_mode`

Reply:
We should remove `self.separate_encoder` for the same reason